src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -0,0 +1,1282 @@
+                    throw new Http2ProtocolException(Http2ProtocolErrorCode.FrameSizeError);
+                }
+
+                // Just eat settings for now
+                // TODO: We should at least validate the settings and fail on invalid settings.
+                // TODO: We should handle SETTINGS_MAX_CONCURRENT_STREAMS.
+                // Others we don't care about, or are advisory.
+                _incomingBuffer.Discard(frameHeader.Length);
+
+                // Send acknowledgement
+                WriteFrameHeader(new FrameHeader(0, FrameType.Settings, FrameFlags.Ack, 0));
+                await FlushOutgoingBytesAsync().ConfigureAwait(false);
+            }
+        }
+
+        private void ProcessPriorityFrame(FrameHeader frameHeader)
+        {
+            Debug.Assert(frameHeader.Type == FrameType.Priority);
+
+            if (frameHeader.StreamId == 0 || frameHeader.Length != FrameHeader.PriorityInfoLength)
+            {
+                throw new Http2ProtocolException(Http2ProtocolErrorCode.ProtocolError);
+            }
+
+            // Ignore priority info.
+
+            _incomingBuffer.Discard(frameHeader.Length);
+        }
+
+        private async Task ProcessPingFrame(FrameHeader frameHeader)
+        {
+            Debug.Assert(frameHeader.Type == FrameType.Ping);
+
+            if (frameHeader.StreamId != 0)
+            {
+                throw new Http2ProtocolException(Http2ProtocolErrorCode.ProtocolError);
+            }
+
+            if (frameHeader.AckFlag)
+            {
+                // We never send PING, so an ACK indicates a protocol error
+                throw new Http2ProtocolException(Http2ProtocolErrorCode.ProtocolError);
+            }
+
+            if (frameHeader.Length != FrameHeader.PingLength)
+            {
+                throw new Http2ProtocolException(Http2ProtocolErrorCode.FrameSizeError);
+            }
+
+            // Send PING ACK
+            WriteFrameHeader(new FrameHeader(FrameHeader.PingLength, FrameType.Ping, FrameFlags.Ack, 0));
+            _outgoingBuffer.EnsureAvailableSpace(FrameHeader.PingLength);
+            _incomingBuffer.ActiveSpan.Slice(0, FrameHeader.PingLength).CopyTo(_outgoingBuffer.AvailableSpan);
+            _outgoingBuffer.Commit(FrameHeader.PingLength);
+            await FlushOutgoingBytesAsync().ConfigureAwait(false);
+
+            _incomingBuffer.Discard(frameHeader.Length);
+        }
+
+        private void ProcessWindowUpdateFrame(FrameHeader frameHeader)
+        {
+            Debug.Assert(frameHeader.Type == FrameType.WindowUpdate);
+
+            if (frameHeader.Length != FrameHeader.WindowUpdateLength)
+            {
+                throw new Http2ProtocolException(Http2ProtocolErrorCode.FrameSizeError);
+            }
+
+            int windowUpdate = (int)((uint)((_incomingBuffer.ActiveSpan[0] << 24) | (_incomingBuffer.ActiveSpan[1] << 16) | (_incomingBuffer.ActiveSpan[2] << 8) | _incomingBuffer.ActiveSpan[3]) & 0x7FFFFFFF);
```

Comment:
Same cache-span-into-a-local nit as earlier.

Reply:
Of course this coudl also use BinaryPrimitives to read the value, so maybe you're saving that for later.  A TODO would be nice in that case.